cmd/kpromo/cmd/pr/pr.go

```diff
@@ -232,78 +267,82 @@ func runPromote(opts *promoteOptions) error {
-			return fmt.Errorf("growing manifest with image filter %s and tag %s: %w", opts.images, opts.tags, err)
+			return false, fmt.Errorf("growing manifest with image filter %s and tag %s: %w", opts.images, opts.tags, err)
 		}
-	}
 
-	// Re-write the image list without the mock images
-	rawImageList, err := image.NewManifestListFromFile(filepath.Join(repo.Dir(), imagesListPath))
-	if err != nil {
-		return fmt.Errorf("parsing the current manifest: %w", err)
-	}
+		// Re-write the image list without the mock images
+		rawImageList, err := image.NewManifestListFromFile(filepath.Join(repo.Dir(), imagesListPath))
+		if err != nil {
+			return false, fmt.Errorf("parsing the current manifest: %w", err)
+		}
 
-	// Create a new imagelist to copy the non-mock images
-	newImageList := &image.ManifestList{}
+		// Create a new imagelist to copy the non-mock images
+		newImageList := &image.ManifestList{}
 
-	// Copy all non mock-images:
-	for _, imageData := range *rawImageList {
-		if !strings.Contains(imageData.Name, "mock/") {
-			*newImageList = append(*newImageList, imageData)
+		// Copy all non mock-images:
+		for _, imageData := range *rawImageList {
+			if !strings.Contains(imageData.Name, "mock/") {
+				*newImageList = append(*newImageList, imageData)
+			}
 		}
-	}
 
-	// Write the modified manifest
-	if err := newImageList.Write(filepath.Join(repo.Dir(), imagesListPath)); err != nil {
-		return fmt.Errorf("while writing the promoter image list: %w", err)
-	}
+		// Write the modified manifest
+		if err := newImageList.Write(filepath.Join(repo.Dir(), imagesListPath)); err != nil {
+			return false, fmt.Errorf("while writing the promoter image list: %w", err)
+		}
 
-	// Check if the image list was modified
-	if len(oldlist) > 0 {
-		logrus.Debug("Checking if the image list was modified")
-		// read the newly modified manifest
-		newlist, err := os.ReadFile(filepath.Join(repo.Dir(), imagesListPath))
-		if err != nil {
-			return fmt.Errorf("while reading the modified manifest images list: %w", err)
+		// Check if the image list was modified
+		if len(oldlist) > 0 {
+			logrus.Debug("Checking if the image list was modified")
+			// read the newly modified manifest
+			newlist, err := os.ReadFile(filepath.Join(repo.Dir(), imagesListPath))
+			if err != nil {
+				return false, fmt.Errorf("while reading the modified manifest images list: %w", err)
+			}
+
+			// If the manifest was not modified, exit now
+			if bytes.Equal(newlist, oldlist) {
+				return false, nil
+			}
 		}
 
-		// If the manifest was not modified, exit now
-		if bytes.Equal(newlist, oldlist) {
-			logrus.Info("No changes detected in the promoter images list, exiting without changes")
-			return nil
+		// add the modified manifest to staging
+		logrus.Debugf("Adding %s to staging area", imagesListPath)
+		if err := repo.Add(imagesListPath); err != nil {
+			return false, fmt.Errorf("adding image manifest to staging area: %w", err)
 		}
```

Comment:
For some reason, this doesn't really feel like part of this function to me. Additionally, if you would put it in the `runPronmote` function, you could avoid passing `repo` and instead pass the value of `repo.Dir()`.